sycamore/scans/file_scan.py

```diff
@@ -226,3 +226,42 @@ def execute(self) -> Dataset:
 
     def format(self):
         return "json"
+
+
+class NestedJsonScan(FileScan):
+    def __init__(
+        self,
+        paths: Union[str, list[str]],
+        *,
+        parallelism: Optional[int] = None,
+        filesystem: Optional[FileSystem] = None,
+        doc_extractor: Optional[Callable] = None,
```

Comment:
Is the only difference from the existing json scan the doc_extractor? Would it be easier to add the doc_extractor as an optional parameter to that scan rather than adding a new one?

Reply:
Yeah that'll actually simplify it, we do a flatmap but I believe the current json _to_document method should work fine with it?